docs/source/guides/build.rst

```diff
@@ -0,0 +1,218 @@
+to install all dependencies with Homebrew.
+
+.. code-block:: shell
+
+  xcode-select --install
+  brew install cmake boost postgres grpc autoconf automake libtool golang libpqxx
+
+.. hint:: To install the Homebrew itself please run 
+
+  ``ruby -e "$(curl -fsSL https://raw.githubusercontent.com/homebrew/install/master/install)"``
+
+Build Process
+-------------
+
+Cloning the Repository
+^^^^^^^^^^^^^^^^^^^^^^
+Clone the `Iroha repository <https://github.com/hyperledger/iroha>`_ to the
+directory of your choice.
+
+.. code-block:: shell
+
+  git clone -b develop https://github.com/hyperledger/iroha
+  cd iroha
+
+.. hint:: If you have installed the prerequisites with Docker, you don't need
+  to clone Iroha again
+
+
+Building Iroha
+^^^^^^^^^^^^^^
+To build Iroha, use those commands
+
+.. code-block:: shell
+
+  mkdir build; cd build; cmake ..; make -j$(nproc)
+
+Alternatively, you can use these shorthand parameters (they are not documented
+though)
+
+.. code-block:: shell
+
+  cmake -H. -Bbuild;
+  cmake --build build -- -j$(nproc)
+
+.. note::  On macOS ``$(nproc)`` variable does not work. Check number of 
+  logical cores with ``sysctl -n hw.ncpu`` and put it explicitly in the command 
+  above, e.g. ``cmake --build build -- -j4``
+
+CMake Parameters
+^^^^^^^^^^^^^^^^
+
+We use CMake to build platform-dependent build files. It has numerous flags 
+for configuring the final build. Note that besides the listed parameters
+cmake's variables can be useful as well. Also as long as this page can be
+deprecated (or just not complete) you can browse custom flags via 
+``cmake -L``, ``cmake-gui``, or ``ccmake``.
+
+.. hint::  You can specify parameters at the cmake configuring stage
+  (e.g cmake -DTESTING=ON).
+
+Main Parameters
+"""""""""""""""
+
++--------------+-----------------+---------+------------------------------------------------------------------------+
+| Parameter    | Possible values | Default | Description                                                            |
++==============+=================+=========+========================================================================+
+| TESTING      |      ON/OFF     | ON      | Enables or disables build of the tests                                 |
++--------------+                 +---------+------------------------------------------------------------------------+
+| BENCHMARKING |                 | OFF     | Enables or disables build of the benchmarks                            |
++--------------+                 +---------+------------------------------------------------------------------------+
+| COVERAGE     |                 | OFF     | Enables or disables coverage                                           |
++--------------+                 +---------+------------------------------------------------------------------------+
+| SWIG_PYTHON  |                 | OFF     | Enables or disables libraries and native interface bindings for python |
```

Comment:
Python

Reply:
… build of library and native interface bindings

Reply:
Enables of disables the library building and Python bindings